src/main/java/net/snowflake/client/core/SecureStorageLinuxManager.java

```diff
@@ -43,58 +44,64 @@ private SecureStorageLinuxManager() {
             .build();
   }
 
-  public static SecureStorageLinuxManager builder() {
-    return new SecureStorageLinuxManager();
+  private static class SecureStorageLinuxManagerHolder {
+    private static final SecureStorageLinuxManager INSTANCE = new SecureStorageLinuxManager();
   }
```

Comment:
Changed SecureStorageLinuxManager to singleton pattern to let threads be able to shared `localCredCache`. Also avoid the problem that different threads may mess up content in local file cache set by other threads.